### ISSUE TITLE
[Feature request] Add shell auto-completion

### ISSUE THREAD

**Issue by barseghyanartur** [explosion/radicli#26]
TL;DR - Similar to how it's implemented in [alternatives](https://click.palletsprojects.com/en/8.1.x/shell-completion/#shell-completion).

**Comment by ines** [comment]
Yes, I've definitely wanted this! 👍 It will depend on other libraries so it could be an option people can activate.

There's a package `argcomplete` that implements auto-completion for `argparse`, so that _should_ work: https://kislyuk.github.io/argcomplete/ But I haven't managed to get it to work yet...
